packages/ui/src/modals/AddMembershipModal/AddMembershipModal.tsx

```diff
@@ -38,12 +41,29 @@ export const AddMembershipModal = ({ onClose }: MembershipModalProps) => {
     [JSON.stringify(transactionParams)]
   )
 
-  const onDone = (result: boolean) => setStep(result ? 'SUCCESS' : 'ERROR')
-
   if (step === 'PREPARE' || !transactionParams) {
     return <MembershipFormModal onClose={onClose} onSubmit={onSubmit} membershipPrice={membershipPrice} />
   }
 
+  const onDone = (result: boolean, events: EventRecord[]) => {
```

Comment:
I'm thinkig if we couldn't move that as before. 

I find the seperation of render function better to read and cleaner. The downside is that you have to make add type sefety checks for `transactionParams` here as well.